lib/mix/lib/mix/compilers/elixir.ex

```diff
@@ -332,9 +342,28 @@ defmodule Mix.Compilers.Elixir do
 
   ## Resolution
 
-  defp update_stale_sources(sources, changed) do
+  defp update_stale_sources(sources, changed, sources_stats \\ %{}) do
     # Store empty sources for the changed ones as the compiler appends data
-    Enum.reduce(changed, sources, &List.keystore(&2, &1, source(:source), source(source: &1)))
+
+    get_size = fn file ->
+      cond do
+        source_stats = Map.get(sources_stats, file) ->
+          elem(source_stats, 1)
+
+        source = List.keyfind(sources, file, source(:source)) ->
+          source(size: size) = source
+          size
+
+        true ->
```

Comment:
Do we ever hit this clause? If so, when? Why? :)

Reply:
Yeah we shouldn't ever hit this clause and it should fail if it ever does :)